builtin/logical/transit/backend.go

```diff
@@ -64,6 +68,28 @@ type backend struct {
 	lm *keysutil.LockManager
 }
 
+// initializeCache initializes a transit's cache
+func (b *backend) initializeCache(ctx context.Context, s logical.Storage) error {
+	// default to an unlimited cache size
+	targetSize := 0
+	// override default size with a stored size value if one is available
+	if entry, _ := s.Get(ctx, "config/cache-size"); entry != nil {
+		var storedCacheSize configCacheSize
+		if err := entry.DecodeJSON(&storedCacheSize); err != nil {
```

Comment:
Before decoding this entry, you might want to check if it's nil (which can happen if you're mounting a fresh backend) and if so, simply call `b.lm.ConvertCacheToSyncmap()` and return.